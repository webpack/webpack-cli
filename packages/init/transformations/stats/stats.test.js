"use strict";

const defineTest = require("@webpack-cli/utils/defineTest");

defineTest(
	__dirname,
	"stats",
	"stats-0",
	{
		assets: true,
		assetsSort: "'field'",
		cached: true,
		cachedAssets: true,
		children: true,
		chunks: true,
		chunkModules: true,
		chunkOrigins: true,
		chunksSort: "'field'",
		context: "'../src/'",
		colors: true,
		depth: false,
		entrypoints: "customVal",
		errors: true,
		errorDetails: true,
		exclude: [],
		hash: true,
		maxModules: 15,
		modules: true,
		modulesSort: "'field'",
		performance: true,
		providedExports: false,
		publicPath: true,
		reasons: true,
		source: true,
		timings: true,
		usedExports: false,
		version: true,
		warnings: true
	},
	"init"
);
defineTest(__dirname, "stats", "stats-0", "'errors-only'", "init");

defineTest(
	__dirname,
	"stats",
	"stats-0",
	{
		assets: true,
		assetsSort: "'naw'",
		cached: true,
		cachedAssets: true,
		children: true,
		chunks: true,
		version: true,
		warnings: false
	},
	"add"
);
defineTest(
	__dirname,
	"stats",
	"stats-1",
	{
		assets: true,
		assetsSort: "'naw'",
		cached: true,
		cachedAssets: true,
		children: true,
		chunks: true,
		version: true,
		warnings: false
	},
	"add"
);
defineTest(__dirname, "stats", "stats-1", "'errors-only'", "add");
