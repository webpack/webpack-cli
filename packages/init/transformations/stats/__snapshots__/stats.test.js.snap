// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`stats transforms correctly using "stats-0" data 1`] = `
"module.exports = {
    entry: 'index.js',

    output: {
		filename: 'bundle.js'
	},

    stats: {
        assets: true,
        assetsSort: 'field',
        cached: true,
        cachedAssets: true,
        children: true,
        chunks: true,
        chunkModules: true,
        chunkOrigins: true,
        chunksSort: 'field',
        context: '../src/',
        colors: true,
        depth: false,
        entrypoints: customVal,
        errors: true,
        errorDetails: true,
        exclude: [],
        hash: true,
        maxModules: 15,
        modules: true,
        modulesSort: 'field',
        performance: true,
        providedExports: false,
        publicPath: true,
        reasons: true,
        source: true,
        timings: true,
        usedExports: false,
        version: true,
        warnings: true
    }
}
"
`;

exports[`stats transforms correctly using "stats-0" data 2`] = `
"module.exports = {
    entry: 'index.js',

    output: {
		filename: 'bundle.js'
	},

    stats: 'errors-only'
}
"
`;

exports[`stats transforms correctly using "stats-0" data 3`] = `
"module.exports = {
    entry: 'index.js',

    output: {
		filename: 'bundle.js'
	},

    stats: {
        assets: true,
        assetsSort: 'naw',
        cached: true,
        cachedAssets: true,
        children: true,
        chunks: true,
        version: true,
        warnings: false
    }
}
"
`;

exports[`stats transforms correctly using "stats-1" data 1`] = `
"module.exports = {
	entry: 'index.js',
	output: {
		filename: 'bundle.js'
	},
	stats: {
        assets: true,
        assetsSort: 'naw',
        cached: true,
        cachedAssets: true,
        children: true,
        chunks: true,
        version: true,
        warnings: false
    }
}
"
`;

exports[`stats transforms correctly using "stats-1" data 2`] = `
"module.exports = {
	entry: 'index.js',
	output: {
		filename: 'bundle.js'
	},
	stats: 'errors-only'
}
"
`;
